app/assets/javascripts/utilities/buildCommentHTML.js.erb

```diff
@@ -46,11 +46,14 @@ function buildCommentHTML(comment) {
     <img class="crayons-avatar__image" width="32" height="32" src="${ comment.user.profile_pic }" alt="${ comment.user.username } profile" />
   </a>`;
 
-  commentHeader = `<header class="comment__header">
-    <a href="/${ comment.user.username }" class="crayons-link crayons-link--secondary flex items-center fw-medium">
+  commentHeader = `<div class="comment__header" >
+    <a href="/${ comment.user.username }" class="crayons-link crayons-link--secondary flex items-center fw-medium m:hidden">
       <span class="js-comment-username">${ comment.user.name }</span>
     </a>
-
+    <div class="profile-preview-card relative mr-4 mb-4 s:mb-0 fw-medium hidden m:block">
+      <button id="comment-profile-preview-trigger-${comment.id}" aria-controls="comment-profile-preview-content-${comment.id}" class="profile-preview-card__trigger px-0 crayons-btn crayons-btn--ghost" aria-label="${comment.user.name} profile details">${comment.user.name}</button>
+      <span data-js-comment-user-id="${comment.user.id}" data-js-dropdown-content-id="comment-profile-preview-content-${comment.id}" class="preview-card-placeholder"></span>
+    </div>
```

Comment:
I really wish we could stop embedding HTML in this JS file :D But that's for another day and another refactoring

Reply:
my experience in this file was not my favourite 🤣

Reply:
@rhymes, is the reason we currently render this chunk of markup in JS related to caching?

Reply:
@nickytonline I think it's so that we don't need to reload the whole article page when a new comment is posted. Instead, when the comment is successfully submitted, we use this `buildCommentHTML` to create the new comment and insert it directly into the DOM